classes/controller/menu/class-enform-post-controller.php

```diff
@@ -0,0 +1,443 @@
+	 */
+	public function display_fields_custom_box() {
+		$list_table = new Enform_Fields_List_Table();
+		$list_table->prepare_items();
+		$list_table->display();
+	}
+
+	/**
+	 * Adds a meta box for the EN questions.
+	 *
+	 * Adds available questions custom meta box to p4en_form edit post page.
+	 */
+	public function add_questions_custom_box() {
+		add_meta_box(
+			'questions_list_box',
+			__( 'Available Questions', 'planet4-engagingnetworks-backend' ),
+			[ $this, 'display_questions_custom_box' ],
+			self::POST_TYPE
+		);
+	}
+
+	/**
+	 * Display questions custom box content.
+	 */
+	public function display_questions_custom_box() {
+		$list_table = new Enform_Questions_List_Table( 'GEN' );
+		$list_table->prepare_items();
+		$list_table->display();
+	}
+
+	/**
+	 * Adds available opt-ins custom meta box to p4en_form edit post page.
+	 */
+	public function add_optins_custom_box() {
+		add_meta_box(
+			'optins_list_box',
+			__( 'Available Opt-ins', 'planet4-engagingnetworks-backend' ),
+			[ $this, 'display_optins_custom_box' ],
+			self::POST_TYPE
+		);
+	}
+
+	/**
+	 * Display opt-ins custom box content.
+	 */
+	public function display_optins_custom_box() {
+		$list_table = new Enform_Questions_List_Table( 'OPT' );
+		$list_table->prepare_items();
+		$list_table->display();
+	}
+
+	/**
+	 * Retrieves data of a specific question/opt-in.
+	 */
+	public function get_supporter_question_by_id() {
+		// If this is an ajax call.
+		if ( wp_doing_ajax() ) {
+			$id                = filter_input( INPUT_GET, 'id', FILTER_SANITIZE_NUMBER_INT );
+			$main_settings     = get_option( 'p4en_main_settings' );
+			$ens_private_token = $main_settings['p4en_private_api'];
+			$ens_api           = new Ensapi( $ens_private_token );
+			$response          = $ens_api->get_supporter_question_by_id( $id );
+
+			wp_send_json( $response );
+		}
+	}
+
+	/**
+	 * Add underscore templates to footer.
+	 */
+	public function print_admin_footer_scripts() {
+		$this->view->view_template( 'selected_enform_fields', [] );
+	}
+
+	/**
+	 * Hook load new page assets conditionally based on current page.
+	 */
+	public function load_assets() {
+		global $pagenow, $typenow;
+		$pages = [
+			'post.php',
+			'post-new.php',
+		];
+
+		// Load assets conditionally using pagenow, typenow on new/edit form page.
+		if ( in_array( $pagenow, $pages, true ) && self::POST_TYPE === $typenow ) {
+			add_action( "load-$pagenow", [ $this, 'load__new_page_assets' ] );
+			add_action( 'admin_print_footer_scripts', [ $this, 'print_admin_footer_scripts' ], 1 );
+		}
+	}
+
+	/**
+	 * Load assets for new/edit form page.
+	 */
+	public function load__new_page_assets() {
+		wp_enqueue_script( 'jquery-ui-core' );
+		wp_enqueue_script( 'jquery-ui-sortable' );
+		wp_enqueue_script( 'jquery-ui-dialog' );
+		wp_enqueue_script( 'jquery-ui-tooltip' );
+		wp_enqueue_style( 'wp-jquery-ui-dialog' );
+		wp_enqueue_style( 'p4en_admin_style_blocks', P4GBKS_ADMIN_DIR . 'css/admin_en.css', [], '0.1' );
```

Comment:
Since https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/302 scripts versions should be replaced by [`\P4GBKS::file_ver( $filepath )`](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/d2fcc534d317dba2c5967deef37d89decbe454a7/classes/class-loader.php#L587).
For `wp_enqueue_script`, you can use [`\P4GBKS::enqueue_local_script('p4en_admin_style_blocks', 'css/admin_en.css')`](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/d2fcc534d317dba2c5967deef37d89decbe454a7/classes/class-loader.php#L607) instead.